src/main/java/racingcar/domain/Car.java

```diff
@@ -0,0 +1,58 @@
+package racingcar.domain;
+
+import java.util.Objects;
+
+public class Car implements Comparable<Car> {
+
+    private static final int INIT_POSITION = 1;
+    private static final int LOWER_MOVABLE_BOUND = 4;
+
+    private Position position = new Position(INIT_POSITION);
+    private final CarName name;
+
+    public Car(CarName name) {
+        this.name = name;
+    }
+
+    public Position getPosition() {
+        return position;
+    }
+
+    public void attemptToMove(int number) {
+        if (isMovable(number)) {
+            this.position = position.move();
+        }
+    }
+
+    private boolean isMovable(int number) {
+        return number >= LOWER_MOVABLE_BOUND;
+    }
+
+    @Override
+    public int compareTo(Car other) {
+        return position.compareTo(other.getPosition());
+    }
+
+    public boolean isSamePositionWith(Car competitor) {
+        return this.position.equals(competitor.position);
+    }
```

Comment:
```suggestion
    public boolean isSamePositionWith(Car competitor) {
        return compoareTo(competitor) == 0
    }
```
상태의 비교가 한방향으로 흐르면 좋겠네요

Reply:
상태의 비교는 한방향으로 흐르게 메모메모!!